src/ol/render/canvas/canvasreplay.js

```diff
@@ -470,6 +471,65 @@ ol.render.canvas.Replay.prototype.replay_ = function(
         }
         ++i;
         break;
+      case ol.render.canvas.Instruction.MOVE_TO_LINE_TO_QUADRATIC_CURVE_TO:
```

Comment:
You are doing all the math at replay time, i.e. it is repeated with every render frame. It would be better to do the math only once, upon replay creation, instead.